src/common/AudioStreamBuilder.cpp

```diff
@@ -186,4 +186,31 @@ Result AudioStreamBuilder::openManagedStream(oboe::ManagedStream &stream) {
     return result;
 }
 
-} // namespace oboe
\ No newline at end of file
+// FIXME Remove this before release. Just for comparison with next method.
+Result AudioStreamBuilder::openSharedStream(std::shared_ptr<AudioStream> &sharedStream) {
+    sharedStream.reset();
+    AudioStream *streamptr;
+    auto result = openStream(&streamptr);
+    if (result == Result::OK) {
+        sharedStream.reset(streamptr);
+        // Save a weak_ptr in the stream for use with callbacks.
+        streamptr->setWeakThis(sharedStream);
+    }
+    return result;
+}
+
+ResultWithValue<std::shared_ptr<AudioStream>> AudioStreamBuilder::openSharedStream() {
```

Comment:
The benefit of this method is that it fully encapsulates stream creation, rather than requiring the caller to create an empty `shared_ptr<AudioStream>` and pass it in as a parameter. It does break from convention though. Slight preference for this method, although happy to go with either one.

Reply:
Let me rescind my "slight preference for `ResultWithValue` method" and replace it with "strong preference for the original `Result` method". All of our existing users (including all our sample code) will be using the "declare and pass stream pointer as parameter" approach. The former method makes it a very easy refactor, the latter does not.